sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSource.scala

```diff
@@ -190,32 +190,31 @@ class FileStreamSource(
     val startTime = System.nanoTime
 
     var allFiles: Seq[FileStatus] = null
-    if (sourceHasMetadata.isEmpty) {
-      if (FileStreamSink.hasMetadata(Seq(path), hadoopConf)) {
-        sourceHasMetadata = Some(true)
-        allFiles = allFilesUsingMetadataLogFileIndex()
-      } else {
-        allFiles = allFilesUsingInMemoryFileIndex()
-        if (allFiles.isEmpty) {
-          // we still cannot decide
+    sourceHasMetadata match {
```

Comment:
simply switched to `sourceHasMetadata match { case... case ... case ...}`
actual diff is quite small